File: test_data/grammar_samples/sample.cs
// C# sample
using System;
using System.Collections.Generic;

namespace Sample
{
    public class Calculator
    {
        public int Add(int a, int b) => a + b;

        public static void Main(string[] args)
        {
            var calc = new Calculator();
            Console.WriteLine($"Result: {calc.Add(2, 3)}");
            var items = new List<string> { "one", "two" };
        }
    }
}

File: test_data/grammar_samples/sample.rs
// Rust sample
use std::collections::HashMap;

fn main() {
    let mut scores: HashMap<&str, i32> = HashMap::new();
    scores.insert("Alice", 100);
    scores.insert("Bob", 85);

    for (name, score) in &scores {
        println!("{}: {}", name, score);
    }

    let result: Result<i32, String> = Ok(42);
    if let Ok(value) = result {
        println!("Got: {}", value);
    }
}
